src/Library.qs

```diff
@@ -88,11 +89,136 @@
         targetRegister : Qubit[]
     )
     : Unit is Adj + Ctl {
-        (ControlledOnInt(address, ApplyPauliFromBitString(PauliX, true, value, _)))
+        (ControlledOnInt(address, ApplyPauliFromBitString(PauliX, true, value,_)))
             (addressRegister!, targetRegister);
     }
 
 ///////////////////////////////////////////////////////////////////////////
 // UTILITY FUNCTIONS
 ///////////////////////////////////////////////////////////////////////////
+
+
+
+//BUCKET-BRIGADE
+///////////////////////////////////////////////////////////////////////////
+// PUBLIC API
+///////////////////////////////////////////////////////////////////////////
+    
+     
+   /// # Summary
+   /// Type representing a generic BBQRAM type.
+   /// # Input
+   /// ## Lookup
+   /// The named operation that will look up data from the BBQRAM. 
+   /// ## AddressSize
+   /// The size (number of bits) needed to represent an address for the QRAM.
+   /// ## DataSize
+   /// The size (number of bits) needed to represent a data value for the QRAM.
+  newtype BBQRAM = (
```

Comment:
This type is missing the read and write named items that will be necessary for making this API consistent with the implicit QROMs. Also should be named QRAM as we want it to be interchangeable for the user once created.